Sources/web3swift/Transaction/AbstractEnvelope.swift

```diff
@@ -0,0 +1,133 @@
+// Package: web3swift
+// Created by Alex Vlasov.
+// Copyright © 2018 Alex Vlasov. All rights reserved.
+//
+// Support for EIP-2718 by Mark Loit 2022
+
+import Foundation
+import BigInt
+
+/// Enumeration for supported transaction types
+/// .legacy for untyped and type 0 transactions EIP155 and older
+/// .eip2930 for type 1 transactions conforming to EIP2930
+/// .eip1559 for type 2 transactions conforming to EIP1559
+public enum TransactionType: UInt, CustomStringConvertible {
+    case legacy
+    case eip2930
+    case eip1559
+
+    case unknown // catch-all type, always place as last position, also used for range checking
+    // only used in parsing, no TransactionObject will ever return this
+
+    public var description: String {
+        switch self {
+        case .legacy: return "Legacy"
+        case .eip2930: return "EIP-2930"
+        case .eip1559: return "EIP-1559"
+        case .unknown: return "Unknown EIP-2718 Type"
+        }
+    }
+}
+
+/// Encoding selector for transaction transmission/hashing or signing
+public enum EncodeType {
+    case transaction
+    case signature
+}
+
+/// Protocol definition for all transaction envelope types
+///  All envelopes must conform to this protocol to work with EthereumTransaction
+///   each implememtation holds all the type specific data
+///   and implments the type specific encoding/decoding
+public protocol AbstractEnvelope: CustomStringConvertible { // possibly add Codable?
+    /// The type of transaction this envelope represents (.legacy, .eip2930, .eip1559)
+    var type: TransactionType { get }
+
+    // common parameters for any transaction
+    /// the nonce value for the transaction
+    var nonce: BigUInt { get set }
+    /// the chainId this transaction is , or will be, signed for
+    var chainID: BigUInt? { get set } // this unfortunately needs to remain optional due to legacy transaction typ
```

Comment:
Suggestion: 
`// this unfortunately needs to remain optional due to legacy transaction typ`

 =>

```
/// Chain ID this transaction is or will be, signed for.
/// Remains optional to support legacy transaction type.
var chainID: BigUInt? { get set }
```

Reply:
Don't like personally such comments ` // ` as no one will see them until they jump into the library source code.
And their location is kind of weird.

Looks like some of them are old and not written by you @mloit . I think this PR is a good place to get rid of some of these comments. Removing them as they get noticed is a good strategy.

Reply:
No that comment is by me... good suggestion for the rewrite. I'm using code-style comments for notes intended for someone looking at the code itself, generally implementation related, and doc-style comments for information that might be useful for someone trying to implement code using the library, so generally usage related. In this case could have gone either way, but you've convinced me and I will rewrite it as a doc comment.